### ISSUE TITLE
Can't see any containers anymore (nor are they started)

### ISSUE THREAD

**Issue by noromamai** [rancher/os#2583]
RancherOS Version: 1.4.0, running on vmware esxi host, installed to disk

Fairly new to this. set this up half a year ago with 1.4.0 with rancher-vmware.iso, installed several stacks and containers. have not had much time since to work on it, so was left unattended, but running. Did check a few times now and then and things were still running. last time about 2 months ago. wanted to log into dashboard today and was greeted with popup asking about joining analytics. and then went to stack and it was empty. tells me to add a host first before i can do anything. after i added the server itself as an agent, i went to infrastructure->containers and saw containers starting/running and also saw my missing containers there, but they slowly disappeared. i found the containers in /var/lib/docker/containers or at least files with config data that look like my containers.

how can i get everything back the way it was?

the esxi server had been rebooted several times, something may have happened the last time. but vmtools is installed so on shutdown the vm should do a clean poweroff.

**Comment by niusmallnan** [comment]
Did you change the docker storage driver?

**Comment by noromamai** [comment]
Nope, I didn't change anything. Only thing that has happened was that the esxi server it runs on has been shutdown and restarted. Maybe I never set it up right and it reverted somehow after reboot? Is there any way to just copy over container directories to a different location so rancher will pick them up again?

**Comment by niusmallnan** [comment]
Can you try `overlay` or `overlay2` to see if containers can come back?
```
ros config set rancher.docker.storage_driver <driver-overlay-or-overlay2>
system-docker restart docker
```

**Comment by noromamai** [comment]
tried both storage drivers and both don't bring back the containers. it seems like it is unable to save the state. because every time i reboot it asks me about joining analytics again. but from command line the disk is mounted and files accessible. label is set corerctly (RANCHER_STATE) on the disk and in the user-config.yml

**Comment by niusmallnan** [comment]
> because every time i reboot it asks me about joining analytics again

Could you provide more details about this?

**Comment by noromamai** [comment]
keep in mind, i am new to rancher(os), this install is the first time i implemented this. i am also new to containers, before rancheros i played around with docker on a synology disk station.

in a nutshell:
- i used the vmware iso (1.4.0) to install rancheros to disk (/dev/sda) on a esxi host.
- installed and configured some containers.
-i did not set up anything concerning storage, assumed everything would be saved onto the virtual disk rancheros is instaleld on.
- the esxi server has been powered down and rebooted
- everything was still working (i think, not 100% sure if the other reboots were during the time i was still configuring the containers)
- didn't have time to continue working on this so it sat for a while
- a few weeks ago the esxi host was powered down to upgrade ram
- last week i wanted to continue working on the containers and didn't see any in rancher anymore.
- when going to the web interface i was greeted with a "Welcome to Rancher" message. I am certain i never got this between reboots before
- i need to re-add the server to itselfs ("add a host", start rancher-agent) after every reboot now
- the first time i started rancher-agent (after the containers disappeared) on the server i saw my containers (and the infrastructure containers) appear in the web interface under infrastructure->containers, but after 2-3 seconds my installed containers disappeared and the infrastructure ones stayed.
- subsequent reboots with starting rancher-agent did not show my containers for those few seconds anymore under infrastructure->containers
- also, after every reboot i am always greeted with the "welcome to rancher" screen on the web interface and i need to add host again and start rancher-agent
- everything IS running off /dev/sda, i confirmed by removing the install iso from the virtual cdrom, just in case it was a stupid boot mistake thing.
- i can still find files belonging to my containers in /var/lib/docker/overlay2 while booted into rancheros 1.4.0
- there are a lot of directories in /var/lib/docker/overlay2, a lot more than there are inside /var/lib/docker/containers/
- inside /var/lib/containers there are only 9 directories, all seem to be from the infrastructure stack

- if the case of the disappearing containers can't be solved, the most important thing would be to rescue the database from the mariadb container that was running. and if possible the environmentvariables of the lost containers.

**Comment by kingsd041** [comment]
@noromamai Can you get the MariaDB container using `docker ps -a` on the host?

**Comment by noromamai** [comment]
thanks for looking into this further, but no, i can't see any mariadb docker running. i have actually already given up on this and have decided to start from scratch again. thanks for all the help.
